api/src/controllers/request.ts

```diff
@@ -6,12 +6,17 @@ export interface UserRequest extends Request {
   files?: any;
 }
 
-// @ts-expect-error params is optinonal
+// @ts-expect-error params is optional
 export interface RouteRequest<T extends BasicRoute> extends UserRequest {
   body: T["payload"];
   query: T["query"];
   params: T["params"];
 }
+export interface ValidatedRouteRequest<T extends BasicRoute> extends RouteRequest<T> {
+  validatedBody: T["payload"];
+  validatedQuery: T["query"];
+  validatedParams: T["params"];
+}
```

Comment:
```suggestion
// @ts-expect-error params is optional
export interface RouteRequest<T extends BasicRoute> extends UserRequest {
  body: T["payload"];
  query: T["query"];
  params: T["params"];
  validatedBody: NonNullable<T["payload"]>;
  validatedQuery: NonNullable<T["query"]>;
  validatedParams: NonNullable<T["params"]>;
}
```
cf requestValidatorMiddleware pour qu'il retourne toujours un object vide si pas de validation